### PR TITLE
Fix running plugin host on electron windows

### PR DESCRIPTION
**Pull request by Tom-Yucw**
#### What it does

Closes https://github.com/eclipse-theia/theia/issues/6846

to run plugins created by yo @theia/plugin on windows.

#### How to test

1.    Using yo @theia/plugin to make a Hello World plug-in
2.    Adding nodejs  version like "@types/node": "12.20.0" in package.json to avoid error "node_modules@types\node\assert.d.ts"
 3.   Run yarn to build plugin
 4.   Run "Hosted Plug-in: Start Instance" in command, and select the source directory which includes file package.json

A new instance is showed up successfully on windows.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/plugin-dev/src/node/hosted-instance-manager.ts [hunk 1]
```diff
@@ -293,6 +293,10 @@ export abstract class AbstractHostedInstanceManager implements HostedInstanceMan
                 }
             };
 
+            if (isWindows) {
+                options.shell = true;
```
Comment (msujew) [inline packages/plugin-dev/src/node/hosted-instance-manager.ts:297]: Do you mind adding a comment on why this change is necessary?

```suggestion
                // Has to be set for running on windows (electron).
                // See also: https://github.com/nodejs/node/issues/3675
                options.shell = true;
```

**Review by msujew — APPROVED** [review]
The changes look good to me. I can confirm that the issue exists on `master` and is addressed by the changes nicely 👍

File: packages/plugin-dev/src/node/hosted-instance-manager.ts (hunk 1 above)
Comment (Tom-Yucw) [reply packages/plugin-dev/src/node/hosted-instance-manager.ts:297]: OK, I will amend this and PR again. 
Thank you very much!

**Comment by msujew** [comment]
@Tom-Yucw Please don't create a new PR for every change, and simply add your changes to this PR. If you choose to amend to your first commit, you can force push them using `git push -f`.

**Comment by vince-fugnitto** [comment]
@msujew should we merge for this week's release?
